### PR TITLE
[Quest API] Resolves traindiscs and scribespells issues in Perl/Lua.

### PR DESCRIPTION
**Pull request by KinglyKrab**
These functions were returning the size of the array of spells, not the amount of spells or disciplines learned.

Was also missing a couple logic checks.

Condensed messages as well and added a count in a message with scribespells similar to traindiscs.

### CODE REVIEW

File: zone/questmgr.cpp [hunk 1]
```diff
@@ -1101,9 +1106,16 @@ uint16 QuestManager::scribespells(uint8 max_level, uint8 min_level) {
 			}
 			initiator->ScribeSpell(spell_id, book_slot);
 			book_slot = initiator->GetNextAvailableSpellBookSlot(book_slot);
+			spells_learned++;
+			spell_learned = true;
 		}
 	}
-	return spell_count;
+
+	if (spell_learned) {
```
Comment (hgtw) [inline zone/questmgr.cpp:1114]: Could just check `spells_learned > 0` here and ditch the extra var

**Review by hgtw — APPROVED** [review]

**Review by thalix1337 — APPROVED** [review]
Verified, traindisc and scribespells now work as expected.

File: zone/questmgr.cpp (hunk 1 above)
Comment (KinglyKrab) [reply zone/questmgr.cpp:1114]: True enough. Will do that for disciplines as well. Variable was just for code legibility.

**Comment by KinglyKrab** [comment]
Added changes based on comments and a change to break loop earlier rather than continuing even if book slot is invalid.

**Review by joligario — APPROVED** [review]

**Comment by KinglyKrab** [comment]
Removed extra space. Please squash when merged.

**Comment by KinglyKrab** [comment]
Merging based on approvals and this is a bug that needs fixed for custom servers.

**Pull request merged**